docs/tutorial_1/1_beam_introduction.ipynb

```diff
@@ -1,876 +1,1681 @@
+    "@dataclass\n",
+    "class MovieTitle:\n",
+    "    movie_id: int\n",
+    "    year: int\n",
+    "    name: str\n",
+    "            \n",
+    "\n",
+    "class MovieViewCoder(beam.coders.Coder):\n",
+    "    \\\"\\\"\\\"Beam Coder Serialiser for our custom MovieView Data Object\\\"\\\"\\\"\n",
+    "    def encode(self, view):\n",
+    "        return (f'{view.user_id}:{view.movie_id}:{view.rating}:{str(view.date.date())}').encode('utf-8')\n",
+    "\n",
+    "    def decode(self, s):\n",
+    "        return MovieView(*s.decode('utf-8').split(':'))\n",
+    "\n",
+    "    def is_deterministic(self):\n",
+    "        return True\n",
+    "\n",
+    "class MovieTitleCoder(beam.coders.Coder):\n",
+    "    \\\"\\\"\\\"Beam Coder Serialiser for custom MovieTitle Data Object\\\"\\\"\\\"\n",
+    "    def encode(self, title):\n",
+    "        # using a separation character which will unlikely appear in titles\n",
+    "        return (f'{title.movie_id}--{title.year}--{title.name}').encode('utf-8')\n",
+    "    \n",
+    "    def decode(self, s):\n",
+    "        return MovieTitle(*s.decode('utf-8').split('--'))\n",
+    "    \n",
+    "    def is_deterministic(self):\n",
+    "        return True\n",
+    "\n",
+    "\n",
+    "# Once defined, we can register the new Coders\n",
+    "beam.coders.registry.register_coder(MovieView, MovieViewCoder)\n",
+    "beam.coders.registry.register_coder(MovieTitle, MovieTitleCoder)\n",
+    "\" > netflix_movies.py"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#@title **Verify** that import from movie_view module works properly { display-mode: \"form\" } \n",
+    "try: \n",
+    "    from netflix_movies import MovieView, MovieTitle\n",
+    "except ImportError:\n",
+    "    print(\"netflix_movies module cannot be imported. Please check!\")\n",
+    "else:\n",
+    "    print(\"MovieView and MovieTitle dataclasses have been successfully imported.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### **`S1.2`** `beam.ParDo` handler\n",
+    "\n",
+    "The [`beam.ParDo`](https://beam.apache.org/releases/pydoc/2.28.0/apache_beam.transforms.core.html#apache_beam.transforms.core.ParDo) processing paradigm is very similar to the `Map` phase of a `Map`/`Shuffle`/`Reduce`-style algorithm. \n",
+    "\n",
+    "The [`ParDo`](https://beam.apache.org/documentation/transforms/python/elementwise/pardo/) transformer is executed in parallel elementwise to the `PCollection`, via custom [`beam.DoFn`](https://beam.apache.org/releases/pydoc/2.28.0/apache_beam.transforms.core.html#apache_beam.transforms.core.DoFn) callable."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@beam.typehints.with_output_types(MovieView)  # type-hint annotation for the output\n",
+    "class ParseMovieViews(beam.DoFn):\n",
```

Comment:
Could you please add
#@title ....
and hide the code by default (it might be too much information for the user)